### PR TITLE
Some intermediate work

### PR DESCRIPTION
**Pull request by MTWhyte**
This is some work in progress from my branch of `libsemigroups_pybind11` which was changing the bindings to accommodate what I think we originally called the 'Todd-Coxeter refactor' changes to `libsemigroups`.

This is a PR to the **wrong base**: this is deliberate in the sense that I won't have time to do the rebase this morning, but hopefully can still avoid the duplication of work since it'll at least be visible what files I've been at here.

### CODE REVIEW

File: libsemigroups_pybind11/__init__.py
Comment (Joseph-Edwards) [inline libsemigroups_pybind11/__init__.py:1]: This file doesn't need to be changed

File: src/bipart.cpp
Comment (Joseph-Edwards) [inline src/bipart.cpp:1]: This file was moved to src/old

File: src/bmat8.cpp
Comment (Joseph-Edwards) [inline src/bmat8.cpp:1]: This file was moved to src/old

File: src/fpsemi-examples.cpp
Comment (Joseph-Edwards) [inline src/fpsemi-examples.cpp:1]: This file was moved to src/old

File: src/froidure-pin.cpp
Comment (Joseph-Edwards) [inline src/froidure-pin.cpp:1]: This file was moved to src/old

File: src/kambites.cpp
Comment (Joseph-Edwards) [inline src/kambites.cpp:1]: This file was moved to src/old

File: src/main.hpp
Comment (Joseph-Edwards) [inline src/main.hpp:1]: This file doesn't need to be changed

File: src/present.cpp
Comment (Joseph-Edwards) [inline src/present.cpp:1]: This files doesn't need to be changed

File: src/word-graph.cpp
Comment (Joseph-Edwards) [inline src/word-graph.cpp:1]: This file doesn't need to be changed

**Comment by MTWhyte** [comment]
Superseded by #147

**Pull request closed without merging**